Client/Frontend/Browser/BrowserLongPressGestureRecognizer.swift

```diff
@@ -0,0 +1,135 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+import UIKit
+
+enum ElementType {
+    case ImageSrc(NSURL)
+    case hrefLink(NSURL)
+}
+
+protocol BrowserLongPressGestureDelegate: class {
+    func browser(browser: Browser, elementsDidLongPressed elements: [ElementType])
+}
+
+class BrowserLongPressGestureRecognizer: UILongPressGestureRecognizer, UIGestureRecognizerDelegate, BrowserHelper {
+    private  weak var browser: Browser!
+    weak var browserLongPressGestureDelegate: BrowserLongPressGestureDelegate?
+
+    override init(target: AnyObject, action: Selector) {
+        super.init(target: target, action: action)
+    }
+
+    required init?(browser: Browser) {
+        super.init()
+        self.browser = browser
+        delegate = self
+        self.minimumPressDuration *= 0.9
+        self.addTarget(self, action: "SELdidLongPress:")
+
+        if let path = NSBundle.mainBundle().pathForResource("LongPress", ofType: "js") {
+            if let source = NSString(contentsOfFile: path, encoding: NSUTF8StringEncoding, error: nil) {
+                var userScript = WKUserScript(source: source, injectionTime: WKUserScriptInjectionTime.AtDocumentStart, forMainFrameOnly: false)
+                self.browser.webView.configuration.userContentController.addUserScript(userScript)
+            }
+        }
+    }
+
+    // MARK: - Gesture Recognizer Delegate Methods
+    func gestureRecognizer(gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWithGestureRecognizer otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        return true
+    }
+
+    // MARK: - Long Press Gesture Handling
+    func SELdidLongPress(gestureRecognizer: UILongPressGestureRecognizer) {
+        if gestureRecognizer.state == UIGestureRecognizerState.Began {
+            //Finding actual touch location in webView
+            var touchLocation = gestureRecognizer.locationInView(self.browser.webView)
+            touchLocation.x -= self.browser.webView.scrollView.contentInset.left
+            touchLocation.y -= self.browser.webView.scrollView.contentInset.top
+            touchLocation.x /= self.browser.webView.scrollView.zoomScale
+            touchLocation.y /= self.browser.webView.scrollView.zoomScale
+
+            evaluateTouchLocation(touchLocation, gestureRecognizer: gestureRecognizer)
+        }
+    }
+
+    /// Check whether long pressed is on valid link. If found valid link then find the rect for link.
+    private func evaluateTouchLocation(touchLocation: CGPoint,gestureRecognizer: UILongPressGestureRecognizer) {
+        self.browser.webView.evaluateJavaScript("findElementsAtPoint(\(touchLocation.x),\(touchLocation.y))", completionHandler:nil)
```

Comment:
Since this is a private one-line function that isn't used anywhere else, you can just inline this into the SELdidLongPress callback above.